directory/directory_dest.go

```diff
@@ -51,20 +52,26 @@ func (d *dirImageDestination) PutBlob(digest string, stream io.Reader) error {
 		}
 	}()
 
-	if _, err := io.Copy(blobFile, stream); err != nil {
-		return err
+	h := sha256.New()
+	tee := io.TeeReader(stream, h)
+
+	size, err := io.Copy(blobFile, tee)
+	if err != nil {
+		return "", -1, err
 	}
 	if err := blobFile.Sync(); err != nil {
-		return err
+		return "", -1, err
 	}
 	if err := blobFile.Chmod(0644); err != nil {
-		return err
+		return "", -1, err
 	}
+	computedDigest := hex.EncodeToString(h.Sum(nil))
```

Comment:
Sure, the directory transport can use whatever names it wants, but `PutBlob` must return a digest using the same format as the other transports so that we can build a manifest in transport-independent code.